src/ipc/Response.h

```diff
@@ -25,23 +26,23 @@ class Response: public RefCountable
 
 public:
     explicit Response(unsigned int aRequestId):
-        requestId(aRequestId) {}
+        requestId(aRequestId), qid(false) {}
+
+    Response(unsigned int aRequestId, const Ipc::QuestionerId &aQid):
+        requestId(aRequestId), qid(aQid) {}
 
     virtual void pack(TypedMsgHdr& msg) const = 0; ///< prepare for sendmsg()
     virtual Pointer clone() const = 0; ///< returns a copy of this
 
-private:
-    Response(const Response&); // not implemented
-    Response& operator= (const Response&); // not implemented
-
 public:
     unsigned int requestId; ///< ID of request we are responding to
+    const QuestionerId qid;
 };
 
 inline
 std::ostream& operator << (std::ostream &os, const Response& response)
 {
-    os << "[response.requestId %u]" << response.requestId << '}';
+    os << "[response.requestId %u]" << response.requestId << " qid: " << response.qid << '}';
```

Comment:
I could not understand what is this printf-like formatting for. This operator was added at 51ea090, but unfortunately, no explanation was provided. Also, this operator seems to be unused (it compiles without it).

Reply:
I bet it is a copy-paste error. See Ipc::Inquirer::status() in commit 51ea090 that you found.

The common-for-status() square brackets should be removed. `%u` dropped, along with self-naming. The missing opening curly brace added. Something along these lines:

```suggestion
    os << "{requestId: " << response.requestId << " qid: " << response.qid << '}';
```

Removing the operator itself is also OK IMO if it is unused. Ideally, its implementation should not be inlined anyway, but that is a bigger out-of-scope change.

Your call.